deeprank2/query.py

```diff
@@ -183,19 +184,30 @@ def build(
         Returns:
             :class:`Graph`: The resulting :class:`Graph` object with all the features and targets.
         """
+        # TODO: Should this be an internal method, as it is only called upon by QueryCollection and not by user?
 
-        try:
-            feature_modules = [importlib.import_module('deeprank2.features.' + name) for name in feature_modules]
-        except TypeError:
-            if isinstance(feature_modules, ModuleType):
-                feature_modules = [feature_modules]
-            elif isinstance(feature_modules, str):
-                feature_modules = [importlib.import_module('deeprank2.features.' + feature_modules)]
+        # TODO: I would prefer if  `_set_feature_modules` would return a list of `ModuleType` instead of `str`,
+        # but this leads to an exception in the pool function in `QueryCollection.process`, that I don't know how to solve.
+        # Tests are currently passing `ModuleType`s directly, which is clearer than passing strings
+        # and is the reason I am allowing both input types.
+
+        feature_modules = [importlib.import_module('deeprank2.features.' + module)
+                            if isinstance(module, str) else module
+                                for module in feature_modules]
         self._pssm_required = conservation in feature_modules
 
-        return self._child_build(feature_modules)
+        # TODO: my gut feeling is that the building can be unified further, but it is
```

Comment:
I think it's fine; it's always a trade off and you've already improved it quite a lot.

Reply:
Yes and no. I think as we start adding more query subclasses, it will be very helpful to have this as unified as possible. But I think we will run into that as we start adding more query types, and can deal with it then. As you said, it is now already much easier to add more classes and understand what the similarities and differences between the current ones are.